### PR TITLE
More jmap logs

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/DelegateSetDeletePerformer.scala [hunk 1]
```diff
@@ -29,17 +30,23 @@ import org.apache.james.jmap.method.DelegateSetDeletePerformer.{DelegateDeletion
...
 
   case class DelegateDeletionFailure(unparsedDelegateId: UnparsedDelegateId, exception: Throwable) extends DelegateDeletionResult {
     def asSetError: SetError = exception match {
-      case e: IllegalArgumentException => SetError.invalidArguments(SetErrorDescription(s"${unparsedDelegateId.id} is not a DelegationId: ${e.getMessage}"))
-      case _ => SetError.serverFail(SetErrorDescription(exception.getMessage))
+      case e: IllegalArgumentException =>
+        LOGGER.info("Illegal arguments while updating a delegation", exception)
+        SetError.invalidArguments(SetErrorDescription(s"${unparsedDelegateId.id} is not a DelegationId: ${e.getMessage}"))
+      case _ =>
+        LOGGER.error("Failed to delete a delegation", exception)
```
Comment (vttranlina) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/DelegateSetDeletePerformer.scala:48]: delete or update?
I see line45 is "updating"

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/DelegatedAccountDeletePerformer.scala [hunk 2]
```diff
@@ -20,26 +20,34 @@
...
   sealed trait DelegatedAccountDeletionResult
 
   case class DelegatedAccountDeletionSuccess(delegationId: DelegationId) extends DelegatedAccountDeletionResult
 
   case class DelegatedAccountDeletionFailure(unparsedDelegateId: UnparsedDelegateId, exception: Throwable) extends DelegatedAccountDeletionResult {
     def asSetError: SetError = exception match {
-      case e: IllegalArgumentException => SetError.invalidArguments(SetErrorDescription(s"${unparsedDelegateId.id} is not a DelegationId: ${e.getMessage}"))
-      case _ => SetError.serverFail(SetErrorDescription(exception.getMessage))
+      case e: IllegalArgumentException =>
+        LOGGER.info("Illegal arguments while updating a delegation", exception)
```
Comment (vttranlina) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/DelegatedAccountDeletePerformer.scala:46]: "updating" => "delete"

**Review by vttranlina — COMMENTED** [review]
How about the add log in only [Method](https://github.com/apache/james-project/blob/1bfc514e074bd8c96020067739a1e4f19f152cc0/server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala#L80) class?
It will be more generic, and easy control

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/DelegateSetDeletePerformer.scala (hunk 1 above)
Comment (Arsnael) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/DelegateSetDeletePerformer.scala:48]: Delete (should modify line 45)

Case class `DelegateDeletionFailure` :)

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/ProblemDetails.scala [hunk 3]
```diff
@@ -41,15 +41,18 @@ object ProblemDetails {
 
   def forThrowable(throwable: Throwable): ProblemDetails = throwable match {
     case exception: IllegalArgumentException =>
+      LOGGER.info("he request was successfully parsed as JSON but did not match the type signature of the Request object: {}", exception.getMessage)
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/ProblemDetails.scala:44]:
```suggestion
      LOGGER.info("The request was successfully parsed as JSON but did not match the type signature of the Request object: {}", exception.getMessage)
```

**Comment by Arsnael** [comment]
```
02:11:23,740 [ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.8_0.11.0:scalafix (scala-check-style) on project james-server-jmap-rfc-8621: Scalafix invoked with errors. Check logs for details. -> [Help 1]
```

**Comment by chibenwa** [comment]
> How about the add log in only [Method](https://github.com/apache/james-project/blob/1bfc514e074bd8c96020067739a1e4f19f152cc0/server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala#L80) class?

Please contribute :-)

**Comment by chibenwa** [comment]
```
10:42:19,389 [ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.8_0.11.0:scalafix (scala-check-style) on project james-server-jmap-rfc-8621: Scalafix invoked with errors. Check logs for details. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.8_0.11.0:scalafix (scala-check-style) on project james-server-jmap-rfc-8621: Scalafix invoked with errors. Check logs for details.
```

**Review by Arsnael — APPROVED** [review]

**Pull request merged**